image-hd.c

```diff
@@ -16,6 +16,7 @@
  */
```

Comment:
[How on earth does one comment on the commit message and not just individual lines of the diff? That's trivial in an email-based workflow]

Strictly speaking the sanity check only ensures that we did not end up writing too much - i.e., the truncate to the expected final size would not catch the case where the file would end up with a smaller total size had we not done that pre-allocation.

Reply:
Yeah, but I was also ok with the first one - just wanted to point it out.